### DISCUSSION TITLE
Graphql java future - now with Spring Graphql

### DISCUSSION THREAD

**Discussion by sagar-rout** [graphql-java/graphql-java#2635] [General]
Hi,

Recently I watched https://www.youtube.com/watch?v=Kq3UhUQdIO8&t=1777s Spring Developer session and Spring and Graphql team worked to create new Spring- Graphql.

May I know what will be the future of graphql-java project ? I got the feeling that now Spring-graphql will be the core and new implementation (feature/bug fixes) will be on top of Spring-graphql and not on graphql-java. I can be completely wrong.

**Comment by andimarek** [comment]
Thanks for your question. I think you got a slightly wrong impression how GraphQL Java and Spring GraphQL relate to each other. Let me try to clarify:

Spring GraphQL is the Spring specific integration of GraphQL Java. This means everything Spring Framework specific is implemented there. For example support for Spring MVC or Spring Data.

GraphQL Java itself (this project) is the general library used by Spring GraphQL. But GraphQL Java  is used by much more than Spring GraphQL. Have a look at https://github.com/graphql-java/awesome-graphql-java for a list of projects using it.

This means both projects (Spring GraphQL and GraphQL Java) will continue to be maintained and will continue to evolve: for example a new general GraphQL feature will be implemented in GraphLQ Java, while Spring GraphQL will take care of all Spring frameworks specifics.

I hope that helps.

**Comment by sagar-rout** [comment]
Thanks a lot for answering it. I think I got confused. It is just graphql-java team won't maintain https://github.com/graphql-java/graphql-java-spring this project for Spring integration and Spring team with graphql java team will maintain this together https://github.com/spring-projects/spring-graphql.

IDK if I should ask it in this discussion or not . Do you guys have any blogpost or session for migration from graphql-java-spring to new spring-graphql. I maybe asking too much.

Below are the dependencies we are using in our graphql server and it seems migration to new spring-graphql will take some effort. I just don't know how much it is ? and also spring-graphql is not GA. I am bit confused.

```
       <dependency>
            <groupId>com.graphql-java</groupId>
            <artifactId>graphql-java-spring-boot-starter-webmvc</artifactId>
            <version>2.0</version>
        </dependency>
        <dependency>
            <groupId>com.graphql-java</groupId>
            <artifactId>java-dataloader</artifactId>
            <version>2.2.1</version>
        </dependency>
```
